src/github3/pulls.py

```diff
@@ -575,6 +576,21 @@ def reviews(self, number=-1, etag=None):
         url = self._build_url('reviews', base_url=self._api)
         return self._iter(int(number), url, PullReview, etag=etag)
 
+    def statuses(self, number=-1, etag=None):
+        """Iterate over the statuses associated with this pull request.
+
+        :param int number:
+            (optional), number of statuses to return. Default: -1 returns all
+            available statuses.
+        :param str etag:
+            (optional), ETag from a previous request to the same endpoint
+        :returns:
+            generator of statuses for this pull request
+        :rtype:
+            :class:`~github3.repos.Status`
+        """
+        return self._iter(number, self.statuses_url, status.Status,  etag=etag)
```

Comment:
I think
```suggestion
        return self._iter(int(number), self.statuses_url, status.Status,  etag=etag)
```
is better because it raises an exception upfront instead of later on in the internals of the `_iter` helper.

(Also testing this new "suggest a change" feature...)

Reply:
New feature works great, applied from mobile!